### PR TITLE
Fix or replace references to PEP 427

### PR DESCRIPTION
**Pull request by sinoroc**
PEP 427 is about the binary distribution format *wheel*. The full standard specification is available in the guide so we should link to it instead of the PEP. The PEP is rather a historical document, PEPs are part of the process, they are not the specification document.

Preview:
* https://python-packaging-user-guide--1229.org.readthedocs.build/en/1229/
* and more specifically:
  * https://python-packaging-user-guide--1229.org.readthedocs.build/en/1229/discussions/wheel-vs-egg/
  * https://python-packaging-user-guide--1229.org.readthedocs.build/en/1229/glossary/#term-Wheel
  * https://python-packaging-user-guide--1229.org.readthedocs.build/en/1229/specifications/platform-compatibility-tags/#faq

See also: https://github.com/pypa/packaging.python.org/pull/1213

### CODE REVIEW

File: source/glossary.rst [hunk 1]
```diff
@@ -237,7 +237,9 @@ Glossary
 
     Wheel
 
-        A :term:`Built Distribution` format introduced by :pep:`427`,
+        A :term:`Built Distribution` format introduced by an official
+        `standard specification
+        <https://packaging.python.org/en/latest/specifications/binary-distribution-format/>`_,
```
Comment (abravalheri) [inline source/glossary.rst:242]: Maybe the following:

```suggestion
        A :term:`Built Distribution` format introduced by an official
        :doc:`standard specification <specifications/binary-distribution-format>`,
```

File: source/discussions/wheel-vs-egg.rst [hunk 2]
```diff
@@ -45,5 +46,5 @@ Here's a breakdown of the important differences between :term:`Wheel` and :term:
 ----
 
 .. [1] Circumstantially, in some cases, wheels can be used as an importable
-       runtime format, although :pep:`this is not officially supported at this time
-       <427#is-it-possible-to-import-python-code-directly-from-a-wheel-file>`.
+       runtime format, although `this is not officially supported at this time
+       <https://packaging.python.org/en/latest/specifications/binary-distribution-format/#is-it-possible-to-import-python-code-directly-from-a-wheel-file>`_.
```
Comment (abravalheri) [inline source/discussions/wheel-vs-egg.rst:50]: Maybe this also work:
```suggestion
       runtime format, although :doc:`this is not officially supported at this time
       <specifications/binary-distribution-format#is-it-possible-to-import-python-code-directly-from-a-wheel-file>`.
```

**Comment by sinoroc** [comment]
@abravalheri Is there anything I can or have to do to get this moving forward? I do not know what the process is for this particular project.

**Comment by abravalheri** [comment]
Hi @sinoroc, I am sorry, I am not a maintainer in this repo 😅
I just noticed the PR and gave it a read...

**Comment by sinoroc** [comment]
@abravalheri I see, thanks, and sorry for the ping :)

**Review by webknjaz — APPROVED** [review]

File: source/specifications/platform-compatibility-tags.rst [hunk 3]
```diff
@@ -274,7 +274,7 @@ Why not use special character <X> rather than ``.`` or ``-``?
     in some contexts (for example, ``+`` must be quoted in URLs, ``~`` is
     used to denote the user's home directory in POSIX), or because the
     advantages weren't sufficiently compelling to justify changing the
-    existing reference implementation for the wheel format defined in :pep:427
+    existing reference implementation for the wheel format defined in :pep:`427`
```
Comment (webknjaz) [inline source/specifications/platform-compatibility-tags.rst:277]: Shouldn't this also be turned into a link?

**Pull request merged**